src/s_audio_jack.c

```diff
@@ -186,11 +187,18 @@ static int jack_xrun(void* arg) {
   return 0;
 }
 
+int max_client_name_size(void) {
+    if (!jack_client_name_size) {
+        return CLIENT_NAME_SIZE_FALLBACK;
+    } else {
+        return jack_client_name_size();
+    }
+}
 
 static char** jack_get_clients(void)
 {
     const char **jack_ports;
-    char* tmp_client_name = (char*)getbytes(jack_client_name_size());
+    char* tmp_client_name = (char*)getbytes(max_client_name_size());
```

Comment:
that seems a bit overcomplicated.
why not just do:

~~~C
char *tmp_client_name = (char*)getbytes(jack_client_name_size?jack_client_name_size():CLIENT_NAME_SIZE_FALLBACK);
~~~

or - if you find this is really unreadable:

~~~C
char *tmp_client_name;
// ...
if(jack_client_name_size)
  tmp_client_name = (char*)getbytes(jack_client_name_size());
else
  tmp_client_name = (char*)getbytes(CLIENT_NAME_SIZE_FALLBACK):
~~~

Reply:
My personal preference is the function, but happy to move it inline if you prefer it.